src/qt/guiutil.cpp

```diff
@@ -354,7 +354,10 @@ boost::filesystem::path static GetAutostartDir()
 
 boost::filesystem::path static GetAutostartFilePath()
 {
-    return GetAutostartDir() / "bitcoin.desktop";
+    std::string chain = ChainNameFromCommandLine();
+    if (chain == CBaseChainParams::MAIN)
+        return GetAutostartDir() / "bitcoin.desktop";
+    return GetAutostartDir() / strprintf("bitcoin-%s.lnk", chain);
```

Comment:
nit: `s/strprintf("bitcoin-%s.lnk", chain) / "bitcoin-"+chain+".lnk"`? But maybe a matter of taste.